src/history.ts

```diff
@@ -7,28 +7,25 @@ import {WalletLike} from './service/wallet/common/wallet.interface';
 import {CoinType} from './service/wallet/types/coin.type';
 
 import {AsyncRouter} from 'express-async-router';
+import {HistoryReqDTO} from './electrum-client/types/history-req-dto';
+import {plainToClass} from 'class-transformer';
+import {validateOrReject} from 'class-validator';
 const asyncRouter = AsyncRouter();
 
 asyncRouter.get('/get_history', async (req: any, res: any) => {
     return await getHistoryHandler(req, res);
 });
 
 async function getHistoryHandler(req: any, res: any) {
-    const coinType = req.query['coinType'];
-    validateCoinType(coinType, res);
-    const page = req.query['page'] || 1;
-    const pageSize = req.query['pageSize'] || 10;
-    const isProd = req.query['isProd'] || false;
+    req.query['page'] = parseInt(req.query['page'] || 1);
+    req.query['pageSize'] = parseInt(req.query['pageSize'] || 10);
 
-    const options: WalletCreateOptionsInterface = {
-        userString: '',
-        isProd: isProd,
-        type: CoinType.BTC,
-        bitcore: null,
-    };
+    const historyReqDTO: HistoryReqDTO = plainToClass(HistoryReqDTO, req.query);
+    await validateOrReject(historyReqDTO);
```

Comment:
const historyReqDTO: HistoryReqDTO = plainToClass(HistoryReqDTO, req.query);
    await validateOrReject(historyReqDTO);

move this to function getHistoryOptions